### ISSUE TITLE
linter: `no-useless-spread` auto-fix is unsafe

### ISSUE THREAD

**Issue by eventualbuddha** [oxc-project/oxc#5405]
As I commented [here](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2018#issuecomment-2324996704), I believe any auto-fix for `no-useless-spread` is potentially invalid. I provided [a specific example](https://github.com/votingworks/vxsuite/blob/720f72ac404bfff011034f275317a02f5d69cff9/libs/pdi-scanner/src/ts/scanner_client.ts#L226-L229) in that comment, but in general I don't believe it's possible to statically determine whether iterating over the original iterable vs. a cloned array of the original iterable will yield different results. Here's a fairly trivial example that results in very different results:

```ts
const set = new Set([1, 2, 3]);

console.log('before:', set);

for (const element of [...set]) {
  console.log('element:', element);
  set.add(element * 2);
}

console.log('after:', set);
```

I don't believe auto-fixes should be made unless the lint rule can, with very high confidence, ensure that the semantics are unchanged.

**Comment by DonIsaac** [comment]
I'm gonna let someone new take this one on, as this is a great first issue.

**Comment by eventualbuddha** [comment]
I can try to take a crack at just removing the autofix.

**Comment by eventualbuddha** [comment]
After looking into this for a bit, it seems my example above is flagged by the rule but not auto-fixed. I think that's probably as it should be. So maybe there's nothing to do here?

**Comment by DonIsaac** [comment]
Checked it out on my side; this is, in fact, not a violation. Closing this ticket as not a bug, and adding test cases for it anyways.
